timeseries/src/test/groovy/com/hurence/logisland/timeseries/timeseries/SaxAnalyserTest.groovy

```diff
@@ -0,0 +1,30 @@
+package com.hurence.logisland.timeseries.timeseries
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import static com.hurence.logisland.timeseries.sax.SaxAnalyzer.anomalyDetect
+import static com.hurence.logisland.timeseries.sax.SaxAnalyzer.saxThreshold
+
+class SaxAnalyserTest extends Specification{
+    @Unroll
+    def "test anomaly detection in sax strings data"(){
+
+        given:
+        def listin = []
+
+        def listout =[]
+        def s1 = "dededdecccdcedeedccddcddccedeeeeeedceefedbdbccdcccp"//usual sensor output
+        def s2 ="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"//  one type character
+        def s3 = "b"//one character output
+        def s4 = "saxsaxsaxsaxsaxsaxsaxsaxsaxsaxsaxsaxsaxsaxsaxsaxsaxh"//pattern anomaly (pattern : sax)
+        def s5 = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaabaaaaaaaaaaaaaaaaa"//one anomaly in one type character output
+        def s6 = "eeffddddddcdffffeeededdddcedcdfdeddcccdcdedcbbbabadcddcddcefeededcddccedddceddKcedcdcbcceeddddcceeded"//unusual output length
+
```

Comment:
Could you add a def s7 = ""
to see what it brings with an empty string ;) ?

Reply:
done already ;)

Reply:
it returns -1